core/server/data/db/connection.js

```diff
@@ -2,6 +2,10 @@ var knex = require('knex'),
     config = require('../../config'),
     knexInstance;
 
+// @TODO:
+// - if you require this file before config file was loaded,
+// - then this file is cached and you have no chance to connect to the db anymore
+// - bring dynamic into this file (db.connect())
```

Comment:
I think the planned restructure of config to use nconf and handle any config setup as part of the install process will resolve this problem.